project/src/ui/code_editor/Tabs.gd

```diff
@@ -0,0 +1,160 @@
+extends Tabs
+
+
+onready var mainControl: Node = get_owner()
+onready var tabs: Tabs = self
+onready var popupWindow = preload("res://src/ui/popup/popup_window.tscn")
+
+class fileinfo:
+	var _index: int
+	var _name: String
+	var _path: String
+	var _content: String
+	var _savedContent: String
+	var _cursorColumn: int
+	var _cursorLine: int
+			
+#Function that Initializes the tabssystem
+func _ready():
+	tabs.add_tab("+")
+	tabs.tab_close_display_policy =Tabs.CLOSE_BUTTON_SHOW_NEVER
+
+#Create a new tab when openning a new file
+func _create_new_tab_with_content(content,path):
+	#Save the content of the current file in memory
+	_save_tab_content()
+	#Get the name of the file
+	var name = path.substr(path.get_base_dir().length()+1,path.length())
+	tabs.remove_tab(tabs.get_tab_count()-1);	#Remove the + tab
+	tabs.add_tab(name)							#Add the actuall tab
+	tabs.add_tab("+")							#Add the + tab
+	var newFile = fileinfo.new()				#Create an instance of fielinfo
+	newFile._index 		= tabs.get_tab_count()-2		
+	newFile._name 		= name
+	newFile._content 	= content
+	newFile._savedContent 	= content
+	newFile._path 		= path
+	newFile._cursorLine = 0
+	newFile._cursorColumn = 0
```

Comment:
create a constructor instead, (you can add args to the _init function) then tis fileinfo.new(arg1, arg2, ....) etc